server/api/runtime_handlers/base.py

```diff
@@ -1631,6 +1637,25 @@ def _get_run_label_selector(project: str, run_uid: str):
             f"{mlrun_constants.MLRunInternalLabels.uid}={run_uid}"
         )
 
+    @staticmethod
+    def _ensure_run_logs_collected(
+        db: DBInterface, db_session: Session, project: str, uid: str, run: dict = None
+    ):
+        log_file_exists, _ = crud.Logs().log_file_exists_for_run_uid(project, uid)
+        if not log_file_exists:
+            # This stays for now for backwards compatibility in case we would not use the log collector but rather
+            # the legacy method to pull logs. It is also a fallback in case the periodic collect job malfunctions.
+            logs_from_k8s = crud.Logs()._get_logs_legacy_method(
+                db_session, project, uid, source=LogSources.K8S, run=run
+            )
+            if logs_from_k8s:
+                logger.info("Storing run logs", project=project, uid=uid)
+                server.api.crud.Logs().store_log(
+                    logs_from_k8s, project, uid, append=False
+                )
+                # Tell the periodic log collection to not request logs (this assumes the run is in terminal state)
+                db.update_runs_requested_logs(db_session, [uid], requested_logs=True)
```

Comment:
1. Why do we assume that the run is in terminal state and not check it?
2. Why do we update the requested_logs only if we have logs from k8s? don't we need to update it also if we doesn't have logs from k8s?

Reply:
1. We already check it outside of the function but I guess we can check it here as well to make sure - added
2. We let the periodic log collection job handle it because it will mark it as requested logs after `max_consecutive_start_log_requests` threshold passed